### DISCUSSION TITLE
Is there a way to hash the asset file names?

### DISCUSSION THREAD

**Discussion by bravecrayon** [maizzle/framework#917] [Q&A]
Replacing an image on the CDN occasionally will be cached at the local level. Is there a recommended way, or can someone point me in a direction, of adding a hash to the asset file names during a build (like `foo-0dc167866d1f.jpeg`) while keeping all of the references in the rendered html correct?

Thanks!

**Comment by cossssmin** [comment]
There isn't a way to hash asset file names yet in Maizzle. And the [URL Parameters](https://maizzle.com/docs/transformers/url-parameters) transformer currently only works on `href` attributes, so that isn't an option either (I'll look into updating that so it works with `src` too).

One option would be to write a custom PostHTML plugin that adds a `?v=...` parameter to image source URLs, for example:

```js
// config.js

module.exports = {
  build: {
    posthtml: {
      plugins: [
        (() => tree => {
          const process = node => {
            if (node.tag === 'img' && node.attrs?.src) {
              const randomNumber = Math.floor(Math.random() * 10 ** 16).toString().padStart(16, '0')
              node.attrs.src = node.attrs.src + `?v=${randomNumber}`
            }

            return node
          }

          return tree.walk(process)
        })()
      ],
    },
    // ...
  },
}
```

Of course that would only work on `<img src="...">`, so if you have other image URLs like in CSS background images, you would need to use a regex instead - you can do that in an event that gives you access to the HTML, for example in [`beforeRender`](https://maizzle.com/docs/events#beforerender).
